src/main/java/duke/command/Command.java

```diff
@@ -0,0 +1,36 @@
+package duke.command;
+
+import duke.exception.DukeException;
+import duke.main.Storage;
+import duke.main.TaskList;
+import duke.main.Ui;
+
+/**
+ * Represents an executable action. A <code>Command</code> object corresponds to
+ * an action to execute as specified by Duke's user.
+ * Determines if program should exit.
+ */
+public abstract class Command {
```

Comment:
I like how the `Command` is created as an abstract class by itself, and its concrete instances are the different command possible. Effective OOP usage.